### PR TITLE
Add CallAutomationAutomatedLiveTestBase and Example Test Cases for CallAutomation Python SDK

### PR DESCRIPTION
**Pull request by atazimsft**
This pull request enables automated live testing in the CallAutomation Python SDK by:

- Introducing the CallAutomationAutomatedLiveTestBase class, which serves as the foundation for creating automated live tests. This class provides core functionality, such as handling events and messages, and managing Service Bus connections.
- Adding the CallAutomationClientAsyncAutomatedLiveTest class that demonstrates how to use the CallAutomationAutomatedLiveTestBase class. This class includes the test_create_VOIP_call_and_answer_then_hangup test case, which simulates a typical call scenario involving creating a VOIP call, answering it, and then hanging up.
- Adding the CallMediaClientAsyncAutomatedLiveTest class that demonstrates how to use the CallAutomationAutomatedLiveTestBase class. This class includes the test_play_media_in_a_call test case, which simulates a typical call scenario involving creating a VOIP call, answering it, playing media, and then hanging up.

These changes provide a robust starting point for developers to create, extend, and maintain automated live tests for the CallAutomation Python SDK.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import os
+import json
+import time
+from datetime import datetime, timedelta
+from typing import Any, Dict, Optional, Type
+
+import requests
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:7]: I wonder if we should keep similar imports together or having them separated as you have. If we keep them separated, should we add comments on why?

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        self.incoming_call_context_store: Dict[str, Any] = {}
+        self.event_store: Dict[str, Dict[str, Any]] = {}
+        super(CallAutomationAutomatedLiveTestBase, self).__init__(method_name, *args, **kwargs)
+    
+    def _format_string(self, s):
+        s1 = f"{s[:12]}-{s[12:16]}-{s[16:20]}-{s[20:24]}-{s[24:36]}"
+        s2 = f"{s[36:44]}-{s[44:48]}-{s[48:52]}-{s[52:56]}-{s[56:]}"
+        return f"{s1}_{s2}"
+    
+    def _parse_ids_from_identifier(self, identifier: str):
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:29]: should we be consistent and mark the methods with the returning type?

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        body = message.body
+        body_bytes = b''.join(body)
+        body_str = body_bytes.decode('utf-8')
+
+        if not body:
+            raise ValueError("Body cannot be empty")
+
+        mapper = json.loads(body_str)
+        
+        if "incomingCallContext" in mapper:
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:44]: is there any other kind of content we'd like to react on?

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+from typing import Any, Dict, Optional, Type
+
+import requests
+from azure.communication.callautomation import CallAutomationEventParser
+from azure.servicebus import ServiceBusClient
+
+from _shared.asynctestcase import AsyncCommunicationTestCase
+
+
+class CallAutomationAutomatedLiveTestBase(AsyncCommunicationTestCase):
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:14]: Please add documentation on the methods to understand what's happening on them.

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py (hunk 1 above)
Comment (atazimsft) [reply sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:7]: I was following the convention as described in the PEP 8 Python style guide
image
https://peps.python.org/pep-0008/

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    def __init__(self, method_name, *args, **kwargs):
+        self.servicebus_connection_str = os.environ.get('SERVICEBUS_STRING', 'Endpoint=sb://REDACTED.servicebus.windows.net/;SharedAccessKeyName=REDACTED;SharedAccessKey=REDACTEDu8EtZn87JJY=')
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:16]: nit: this last part of the sharedkey, was this supposed to be like this or the whole thing should be redacrted?

File: sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        self.dispatcher_endpoint = os.environ.get('DISPATCHER_ENDPOINT', 'https://incomingcalldispatcher.azurewebsites.net')
```
Comment (minwoolee-msft) [inline sdk/communication/azure-communication-callautomation/tests/call_automation_automated_live_test_base.py:17]: We probably want to not include default dispatcher endpoint by default

**Review by juntuchen-msft — APPROVED** [review]

**Review by juntuchen-msft — APPROVED** [review]

**Pull request merged**